File: src/main/java/com/tatsuaki/carepreventioncsv/model/domain/CarePreventionCsvColumn/ImplementationDivisionForTarget.java
package com.tatsuaki.carepreventioncsv.model.domain.CarePreventionCsvColumn;

import java.util.Arrays;
import java.util.List;

import static java.util.stream.Collectors.toList;

public class ImplementationDivisionForTarget {

    private final String implementationDivisionForTarget;

    public enum Code {

        TARGET("1", "実施不可"),
        NON_TARGET("2", "実施可");

        private String code;
        private String japanese;

        Code(String code, String japanese) {
            this.code = code;
            this.japanese = japanese;
        }

        public String getCode() {
            return code;
        }

        public String getJapanese() {
            return japanese;
        }

        public static boolean contains(String targetCode) {
            List<ImplementationDivisionForTarget.Code> targetCodes = Arrays.stream(values())
                    .filter( value -> value.getCode().equals(targetCode) )
                    .limit(1)
                    .collect(toList());

            return targetCodes.size() != 0;
        }

        public static String getJapanese(String targetCode) {
            List<ImplementationDivisionForTarget.Code> targetCodes = Arrays.stream(values())
                    .filter( value -> value.getCode().equals(targetCode) )
                    .limit(1)
                    .collect(toList());

            if (targetCodes.size() == 0) {
                return "";
            }

            return targetCodes.get(0).getJapanese();
        }
    }

    public ImplementationDivisionForTarget(String implementationDivisionForTarget) {
        this.implementationDivisionForTarget = implementationDivisionForTarget;
    }

    public String getImplementationDivisionForTarget() {
        return implementationDivisionForTarget + getJapanese(implementationDivisionForTarget);
    }

    private String getJapanese(String implementationDivisionForTarget) {
        return validateCodeFormat() ?
                "(" + ImplementationDivisionForTarget.Code.getJapanese(implementationDivisionForTarget) + ")" : "";
    }

    public String getFormatErrorMessage() {
        StringBuilder errorMessageBuilder = new StringBuilder();

        if (!validateCodeFormat()) {
            errorMessageBuilder.append("存在しない事業対象者実施区分コードです.");
        }

        return errorMessageBuilder.toString();
    }

    private boolean validateCodeFormat() {
        return ImplementationDivisionForTarget.Code.contains(implementationDivisionForTarget);
    }
}

File: src/main/java/com/tatsuaki/carepreventioncsv/model/domain/CarePreventionCsvColumn/ImplementationDivisionForTarget.kt
package com.tatsuaki.carepreventioncsv.model.domain.CarePreventionCsvColumn

import java.util.Arrays

import java.util.stream.Collectors.toList

class ImplementationDivisionForTarget(private val implementationDivisionForTarget: String) {

    val formatErrorMessage: String
        get() {
            val errorMessageBuilder = StringBuilder()

            if (!validateCodeFormat()) {
                errorMessageBuilder.append("存在しない事業対象者実施区分コードです.")
            }

            return errorMessageBuilder.toString()
        }

    enum class Code private constructor(val code: String, val japanese: String) {

        TARGET("1", "実施不可"),
        NON_TARGET("2", "実施可");


        companion object {

            operator fun contains(targetCode: String): Boolean {
                val targetCodes = Arrays.stream(values())
                        .filter { value -> value.code == targetCode }
                        .limit(1)
                        .collect(toList())

                return targetCodes.size != 0
            }

            fun getJapanese(targetCode: String): String {
                val targetCodes = Arrays.stream(values())
                        .filter { value -> value.code == targetCode }
                        .limit(1)
                        .collect(toList())

                return if (targetCodes.size == 0) {
                    ""
                } else targetCodes[0].japanese

            }
        }
    }

    fun getImplementationDivisionForTarget(): String {
        return implementationDivisionForTarget + getJapanese(implementationDivisionForTarget)
    }

    private fun getJapanese(implementationDivisionForTarget: String): String {
        return if (validateCodeFormat())
            "(" + ImplementationDivisionForTarget.Code.getJapanese(implementationDivisionForTarget) + ")"
        else
            ""
    }

    private fun validateCodeFormat(): Boolean {
        return ImplementationDivisionForTarget.Code.contains(implementationDivisionForTarget)
    }
}
